components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/utils/SecondaryUserStoreConfigurationUtil.java

```diff
@@ -343,10 +344,17 @@ public static String getUserStoreProperties(UserStoreDTO userStoreDTO, String ex
 
         Map<String, String> secondaryUserStoreProperties = null;
         try {
-            UserStoreManager secondaryUserStoreManager = getSecondaryUserStoreManager(userStoreDomain);
-            if (secondaryUserStoreManager != null) {
-                secondaryUserStoreProperties = secondaryUserStoreManager.getRealmConfiguration()
-                        .getUserStoreProperties();
+            RealmConfiguration realmConfiguration = UserStoreConfigComponent.getRealmService().getTenantUserRealm(
+                    getTenantIdInTheCurrentContext()).getRealmConfiguration();
+            while (realmConfiguration != null) {
+                String domainName = realmConfiguration.getUserStoreProperty(UserCoreConstants.RealmConfig
+                        .PROPERTY_DOMAIN_NAME);
+                if (StringUtils.equalsIgnoreCase(domainName, userStoreDomain)) {
+                    secondaryUserStoreProperties = realmConfiguration.getUserStoreProperties();
+                    break;
+                } else {
```

Comment:
```suggestion
                                  realmConfiguration = realmConfiguration.getSecondaryRealmConfig();
```
Shall we remove the `else` part, as we breaks the flow once the domains are equal.

Reply:
we need to else part to recursively check on all userstores